webapp/src/blocks.tsx

```diff
@@ -749,12 +759,13 @@ export class Editor extends srceditor.Editor {
         blocks.filter(b => b.isShadow_).forEach(b => b.dispose(false));
     }
 
-    private getBlocklyOptions(showCategories?: CategoryMode) {
+    private getBlocklyOptions(showCategories?: boolean) {
         let blocklyOptions = this.getDefaultOptions();
         Util.jsonMergeFrom(blocklyOptions, pxt.appTarget.appTheme.blocklyOptions || {});
-        const hasCategories = showCategories ? showCategories !== CategoryMode.None :
-            (blocklyOptions.hasCategories != undefined ? blocklyOptions.hasCategories : this.showToolboxCategories);
+        const hasCategories = showCategories != undefined ? showCategories :
```

Comment:
Checking if a Boolean is undefined?

Reply:
differentiating between undefined and false.

Reply:
Nit: is using `typeof hasCategories === "undefined"` more explicit?